ArmPkg/Library/ArmFfaLib/ArmFfaCommon.c

```diff
@@ -0,0 +1,758 @@
+/** @file
+  Arm Ffa library common code.
+
+  Copyright (c) 2024, Arm Limited. All rights reserved.<BR>
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+   @par Glossary:
+     - FF-A - Firmware Framework for Arm A-profile
+
+   @par Reference(s):
+     - Arm Firmware Framework for Arm A-Profile [https://developer.arm.com/documentation/den0077/latest]
+
+**/
+#include <Uefi.h>
+#include <Pi/PiMultiPhase.h>
+
+#include <Library/ArmLib.h>
+#include <Library/ArmFfaLib.h>
+#include <Library/ArmSmcLib.h>
+#include <Library/ArmSvcLib.h>
+#include <Library/BaseLib.h>
+#include <Library/BaseMemoryLib.h>
+#include <Library/DebugLib.h>
+#include <Library/PcdLib.h>
+
+#include <IndustryStandard/ArmFfaSvc.h>
+#include <IndustryStandard/ArmFfaPartInfo.h>
+
+#include "ArmFfaCommon.h"
+
+BOOLEAN  gFfaSupported;
+UINT16   gPartId;
+
+/**
+  Convert EFI_GUID to UUID format.
+  for example, If there is EFI_GUID named
+  "378daedc-f06b-4446-8314-40ab933c87a3",
+
+  EFI_GUID is saved in memory like:
+     dc ae 8d 37
+     6b f0 46 44
+     83 14 40 ab
+     93 3c 87 a3
+
+  However, UUID should be saved like:
+     37 8d ae dc
+     f0 6b 44 46
+     83 14 40 ab
+     93 3c 87 a3
+
+  FF-A and other software components (i.e. linux-kernel)
+  uses below format.
+
+  @param [in] Guid            EFI_GUID
+  @param [out] Uuid           Uuid
+
+**/
+STATIC
+VOID
+EFIAPI
+ConvertEfiGuidToUuid (
+  IN   EFI_GUID  *Guid,
+  OUT  UINT64    *Uuid
+  )
+{
+  UINT32  *Data32;
+  UINT16  *Data16;
+
+  CopyGuid ((EFI_GUID *)Uuid, Guid);
+  Data32    = (UINT32 *)Uuid;
+  Data32[0] = SwapBytes32 (Data32[0]);
+  Data16    = (UINT16 *)&Data32[1];
+  Data16[0] = SwapBytes16 (Data16[0]);
+  Data16[1] = SwapBytes16 (Data16[1]);
+}
+
+/**
+  Convert EFI_STATUS to FFA return code.
+
+  @param [in] Status          edk2 status code.
+
+  @retval ARM_FFA_RET_*       return value correspond to EFI_STATUS.
+
+**/
+UINTN
+EFIAPI
+EfiStatusToFfaStatus (
+  IN EFI_STATUS  Status
+  )
+{
+  switch (Status) {
+    case EFI_SUCCESS:
+      return ARM_FFA_RET_SUCCESS;
+    case EFI_INVALID_PARAMETER:
+      return ARM_FFA_RET_INVALID_PARAMETERS;
+    case EFI_OUT_OF_RESOURCES:
+      return ARM_FFA_RET_NO_MEMORY;
+    case EFI_NO_RESPONSE:
+      return ARM_FFA_RET_BUSY;
+    case EFI_INTERRUPT_PENDING:
+      return ARM_FFA_RET_INTERRUPTED;
+    case EFI_ACCESS_DENIED:
+      return ARM_FFA_RET_DENIED;
+    case EFI_ABORTED:
+      return ARM_FFA_RET_ABORTED;
+    case EFI_NOT_FOUND:
+      return ARM_FFA_RET_NODATA;
+    case EFI_NOT_READY:
+      return ARM_FFA_RET_NOT_READY;
+    default:
+      return ARM_FFA_RET_NOT_SUPPORTED;
+  }
+}
+
+/**
+  Convert FFA return code to EFI_STATUS.
+
+  @param [in] FfaStatus          Ffa return Status
+
+  @retval EFI_STATUS             return value correspond EFI_STATUS to FfaStatus
+
+**/
+EFI_STATUS
+EFIAPI
+FfaStatusToEfiStatus (
+  IN UINTN  FfaStatus
+  )
+{
+  switch (FfaStatus) {

```

Comment:
Maybe this is lost on the rebasing. But should this be?
```suggestion
  switch ((INT32)FfaStatus) {
```